src/server/saved_objects/service/lib/repository.js

```diff
@@ -54,36 +56,47 @@ export class SavedObjectsRepository {
    * @param {object} [options={}]
    * @property {string} [options.id] - force id on creation, not recommended
    * @property {boolean} [options.overwrite=false]
+   * @property {object} [options.migrationVersion=undefined]
    * @returns {promise} - { id, type, version, attributes }
   */
   async create(type, attributes = {}, options = {}) {
     const {
       id,
-      overwrite = false
+      migrationVersion,
+      overwrite = false,
     } = options;
 
     const method = id && !overwrite ? 'create' : 'index';
     const time = this._getCurrentTime();
 
     try {
+      const migrated = this._migrator.migrateDocument({
```

Comment:
question: Why do we need to deal with document transformations in `create`? I can understand `bulkCreate` since we need that from the UI import stuff.

Reply:
I think there was a path through which create was used for imports, rather than bulk create, but I could be mistaken. This might just be defensive programming.

Reply:
++ on leaving the validation, I think that's an awesome change. But we should remove the migration stuff if possible so we're not exposing a feature we don't really want people to use. I can't think of a reason off-hand why import would use `create`, but I wouldn't rule anything out with that crazy UI import stuff. If we are, perhaps we'll get lucky and we could just swap it out with a bulkCreate call instead. It's hacky, but that whole import thing is hacky.

Reply:
Ah. I just started looking into changing this, and I remembered the reason this is here. We need *all* docs in the index to have an up to date `migrationVersion`, or else migrations will run next time Kibana boots. So, any time we write a new document, we need to slap the appropriate`migrationVersion` onto it. That's what the migrator is in effect doing here. It also happens to properly handle the case where it may be used as an import, but that's secondary. We could explicitly change this from `migrateDocument` to something more indicative of this, such as `setMigrationVersion` or something. Naming is hard. At the very least, I'll add a comment about why we're doing this and what to keep in mind if we ever want to remove it, and I'll double-check that we've got tests around this.

Reply:
I think something like the setMigrationVersion approach would be better so we get the proper migration version without the side effect of actually migrating, but I'm fine with postponing that change until some other time.